RELEASES.md

```diff
@@ -0,0 +1,254 @@
+considered "pre-releases".
+
+### Major and Minor Releases
+
+Major and minor releases of containerd will be made from master. Releases of
+containerd will be marked with GPG signed tags and announced at
+https://github.com/containerd/containerd/releases. The tag will be of the
+format `v<major>.<minor>.<patch>` and should be made with the command `git tag
+-s v<major>.<minor>.<patch>`.
+
+After a minor release, a branch will be created, with the format
+`release/<major>.<minor>` from the minor tag. All further patch releases will
+be done from that branch. For example, once we release `v1.0.0`, a branch
+`release/1.0` will be created from that tag. All future patch releases will be
+done against that branch.
+
+### Pre-releases
+
+Pre-releases, such as alphas, betas and release candidates will be conducted
+from their source branch. For major and minor releases, these releases will be
+done from master. For patch releases, these pre-releases should be done within
+the corresponding release branch.
+
+While pre-releases are done to assist in the stabilization process, no
+guarantees are provided.
+
+### Upgrade Path
+
+The upgrade path for containerd is such that the 0.0.x patch releases are
+always backward compatible with its major and minor version. Minor (0.x.0)
+version will always be compatible with the previous minor release. i.e. 1.2.0
+is backwards compatible with 1.1.0 and 1.1.0 is compatible with 1.0.0. There is
+no compatibility guarantees for upgrades that span multiple, _minor_ releases.
+For example, 1.0.0 to 1.2.0 is not supported. One should first upgrade to 1.1,
+then 1.2.
+
+There are no compatibility guarantees with upgrades to _major_ versions. For
+example, upgrading from 1.0.0 to 2.0.0 may require resources to migrated or
+integrations to change. Each major version will be supported for at least 1
+year with bug fixes and security patches.
+
+### Next Release
+
+The activity for the next release will be tracked in the
+[milestones](https://github.com/containerd/containerd/milestones). If your
+issue or PR is not present in a milestone, please reach out to the maintainers
+to create the milestone or add an issue or PR to an existing milestone.
+
+### Support Horizon
+
+We'll discuss the support horizon a `minor` release in the context of the
+corresponding release branch, identified by `<major>.<minor>`. Releases
+branches will be in one of several states:
+
+- __*Next*__: The next planned release branch.
+- __*Active*__: The release is currently supported and accepting patches.
+- __*End of Life*__: The release branch is no longer support and no new patches will be accepted.
+
+Releases will be supported up to one year after a _minor_ release. This means that
+we will accept bug reports and backports to release branches until the end of
+life date. If no new _minor_ release has been made, that release will be
+considered supported until the next _minor_ is released or one year, whichever
+is longer.
```

Comment:
You may want to consider some (brief) period of overlap in support in case issues arise in the upgrade path between minor versions that are best addressed with patches to the previous (otherwise unsupported) release. Or maybe just call out the possibility that exceptions can be made in situations that interfere with a clean upgrade.